### ISSUE TITLE
Module system breaks zip support

### ISSUE THREAD

**Issue by chaigler** [GarageGames/Torque3D#2176]
The new module system pulled in from T2D fails to load modules from a zipped archive. Console log shows something similar to:

> Module Manager: Cannot load explicit module Id 'CoreComponentsModule' as it does not exist.

This occurs on the latest development branch using the new (BaseGame) template. 

The odd thing is a game will still work if the 'data' folder is left out of the zip (only 'core' and 'tools' go in game.zip) despite the warnings about not being able to load the Core & Tools modules. Putting the 'data' folder in the zip just results in the engine running with a blank screen.

**Comment by Areloch** [comment]
Yeah, the module scan logic doesn't currently interface with the virtual folders. it's a known bug that'll definitely get fixed.

Thanks for the issue for keep tabs on it thought! :)
